pkg/cmd/workspace/code.go

```diff
@@ -111,6 +109,16 @@ var CodeCmd = &cobra.Command{
 			}
 		}
 
+		project := workspace.GetProjectByName(projectName)
+		if project != nil {
```

Comment:
You can preform this check above in the for loop.

Reply:
the above loop loops through `workspace.Info.ProjectInfo`, i had tried checking for the IsRunning status, which doesn't work.

on the other hand, looping through the `workspace.projects` to find the project and then checking the state of it works.